test/e2e/syncer/syncer_test.go

```diff
@@ -251,6 +255,76 @@ func TestSyncerLifecycle(t *testing.T) {
 			return true
 		}, wait.ForeverTestTimeout, time.Millisecond*100, "upstream configmap %s/%s was not found", upstreamNamespace.Name, expectedConfigMapName)
 
+		// Tunnels KCP <----> Syncer
+		t.Logf("Create upstream service account permissions for downstream in-cluster tunnel pods access")
+		podsAllRole := &rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{Name: "pods-all"},
+			Rules: []rbacv1.PolicyRule{
+				rbachelper.NewRule("*").Groups("").Resources("pods").RuleOrDie(),
+				rbachelper.NewRule("*").Groups("").Resources("pods/log").RuleOrDie(),
+				rbachelper.NewRule("*").Groups("").Resources("pods/exec").RuleOrDie(),
+			},
+		}
+
+		_, err = upstreamKubeClusterClient.RbacV1().Roles(upstreamNamespace.Name).Create(logicalcluster.WithCluster(ctx, wsClusterName), podsAllRole, metav1.CreateOptions{})
+		require.NoError(t, err, "failed to create upstream role")
+		podsAllRoleBinding := &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{Name: "pods-all"},
+			Subjects: []rbacv1.Subject{
+				{Kind: "ServiceAccount", Name: syncerFixture.SyncerID},
+			},
+			RoleRef: rbacv1.RoleRef{Kind: "ClusterRole", Name: "pods-all"},
+		}
+
+		_, err = upstreamKubeClusterClient.RbacV1().RoleBindings(upstreamNamespace.Name).Create(logicalcluster.WithCluster(ctx, wsClusterName), podsAllRoleBinding, metav1.CreateOptions{})
+		require.NoError(t, err, "failed to create upstream rolebinding")
+
+		// KCP will forwards all requests to the downstream cluster
+		// kubectl --server=https://{host}/clusters/{cluster}/services/tunnels/syncer-proxy/{syncer-name} get pods -A
+		t.Logf("Get logs through KCP from downstream deployment")
+		u, err := url.Parse(syncerFixture.SyncerConfig.UpstreamConfig.Host)
+		require.NoError(t, err, "failed to parse upstream Host for syncer")
+		proxiedConfig := rest.CopyConfig(syncerFixture.SyncerConfig.UpstreamConfig)
+		proxiedConfig.Host = fmt.Sprintf("%s://%s/clusters/%s/services/tunnels/syncer-proxy/%s", u.Scheme, u.Host, wsClusterName, syncerFixture.SyncerConfig.SyncTargetName)
+
+		proxiedKcpClient, err := kubernetesclientset.NewForConfig(proxiedConfig)
+		require.NoError(t, err)
+
+		framework.Eventually(t, func() (bool, string) {
+			deployment, err = proxiedKcpClient.AppsV1().Deployments(downstreamNamespaceName).Get(ctx, iccUpstreamDeployment.Name, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				return false, ""
```

Comment:
return the error as string: `return false, fmt.Sprintf("some context: %v", err)`